OpenSim/Common/Storage.cpp

```diff
@@ -2287,28 +2287,24 @@ pad(int aPadSize)
     delete[] vecs;
 }
 
-//_____________________________________________________________________________
-/**
- * Smooth spline each of the columns in the storage.  Note that as a part
- * of this operation, the storage is resampled so that the statevectors are
- * at equal spacing.
- *
- * @param aOrder Order of the spline.
- * @param aCutoffFrequency Cutoff frequency.
- */
 void Storage::
 smoothSpline(int aOrder,double aCutoffFrequency)
 {
+    int size = getSize();
     double dtmin = getMinTimeStep();
+    double avgDt = (_storage[size - 1].getTime() - _storage[0].getTime()) / size;
```

Comment:
I think you need to divide by `size-1` (because it takes only 19 cuts to get 20 pieces of bread from a loaf). For example, suppose the time column is [0, 0.1, 0.2, 0.3, 0.4, 0.5]. The minimum step size is 0.1 but `avgDt` will be calculated as (0.5-0)/6 = 0.083333. Not sure why this off-by-one error wasn't caught by a test.

Also see L2335 and L2372.